metagpt/schema.py

```diff
@@ -4,32 +4,145 @@
 @Time    : 2023/5/8 22:12
 @Author  : alexanderwu
 @File    : schema.py
+@Modified By: mashenquan, 2023-10-31. According to Chapter 2.2.1 of RFC 116:
+        Replanned the distribution of responsibilities and functional positioning of `Message` class attributes.
+@Modified By: mashenquan, 2023/11/22.
+        1. Add `Document` and `Documents` for `FileRepository` in Section 2.2.3.4 of RFC 135.
+        2. Encapsulate the common key-values set to pydantic structures to standardize and unify parameter passing
+        between actions.
+        3. Add `id` to `Message` according to Section 2.2.3.1.1 of RFC 135.
 """
 from __future__ import annotations
 
-from dataclasses import dataclass, field
-from typing import Type, TypedDict
+import asyncio
+import json
+import os.path
+import uuid
+from asyncio import Queue, QueueEmpty, wait_for
+from json import JSONDecodeError
+from pathlib import Path
+from typing import Dict, List, Optional, Set, TypedDict
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
 
+from metagpt.config import CONFIG
+from metagpt.const import (
+    MESSAGE_ROUTE_CAUSE_BY,
+    MESSAGE_ROUTE_FROM,
+    MESSAGE_ROUTE_TO,
+    MESSAGE_ROUTE_TO_ALL,
+    SYSTEM_DESIGN_FILE_REPO,
+    TASK_FILE_REPO,
+)
 from metagpt.logs import logger
+from metagpt.utils.common import any_to_str, any_to_str_set
 
 
 class RawMessage(TypedDict):
     content: str
     role: str
 
 
-@dataclass
-class Message:
+class Document(BaseModel):
```

Comment:
what's the difference with `Document` in metagpt/document.py and `Document` in metagpt/document_store/document.py

Reply:
- `Document` in `shema.py` is a data structure in memory used to point to files in the file system.
- `metagpt/document.py` represents files in the file system.
- `metagpt/document_store/document.py` represents files in the vector database.

Reply:
maybe with different class name is more clear.